arvo/eyre/external-api-ref.md

```diff
@@ -0,0 +1,399 @@
+
+If successful, Eyre will respond with a status code of 204. Note the HTTP response will have no content, any [responses](#responses) will be sent as SSEs on the channel event stream.
+
+## Actions
+
+### Poke
+
+This is for poking a Gall agent.
+
+| Key      | JSON Type | Example Value | Description                                                   |
+|----------|-----------|---------------|---------------------------------------------------------------|
+| `id`     | Number    | `1`           | ID for keeping track of sent messages.             |
+| `action` | String    | `'poke'`      | The kind of action.                                           |
+| `ship`   | String    | `'zod'`       | Target ship name, excluding leading `~`.                             |
+| `app`    | String    | `'hood'`      | Name of the gall agent you're poking.                         |
+| `mark`   | String    | `'helm-hi'`   | Type of data. Must correspond to a mark definition in `/mar`. |
+| `json`   | Any       | `'hello'`     | Actual payload. Any JSON type, determined by app.             |
+
+**Example**
+
+```json
+{
+  'id': 1,
+  'action': 'poke',
+  'ship': 'zod',
+  'app': 'hood',
+  'mark': 'helm-hi',
+  'json': 'hello'
+}
+```
+
+**Response**
+
+Eyre will send a [poke ack](#poke-ack) as an SSE event on the channel event stream.
+
+### Subscribe
+
+This is for subscribing to a watch `path` of a Gall agent.
+
+| Key      | JSON Type | Example Value   | Description                                         |
+|----------|-----------|-----------------|-----------------------------------------------------|
+| `id`     | Number    | `2`             | ID for keeping track of sent messages.   |
+| `action` | String    | `'subscribe'`   | The kind of action.                                 |
+| `ship`   | String    | `'zod'`         | Target ship name, excluding leading `~`.                   |
+| `app`    | String    | `'graph-store'` | Name of the gall agent to which you're subscribing. |
+| `path`   | String    | `'/updates'`    | The path to watch. Depends on the app.              |
+
+**Example**
+
+```json
+{
+  'id': 2,
+  'action': 'subscribe',
+  'ship': 'zod',
+  'app': 'graph-store',
+  'path': '/updates'
+}
+```
+
+**Response**
+
+Eyre will send back a [watch ack](#watch-ack). If subscribing was successful, you will also begin receiving any [diff](#diff)s sent by the Gall agent on the specified `path`.
+
+### Ack
+
+This is for acknowledging an SSE event. If you `ack` one event, you also implicitly `ack` all previous events. Events must be `ack`ed.
```

Comment:
Either here or in the examples (or both?) might be worth mentioning explicitly that they must be acked _so that Eyre can forget them_, and that if you don't, Eyre will eventually kill your subscription. This "clogging" behavior is mentioned briefly in the data-types doc, but should probably be elaborated somewhere.